contracts/Core/StakeManager.sol

```diff
@@ -392,7 +395,8 @@ contract StakeManager is Initializable, ACL, StakeStorage {
         uint256 _stake,
         string memory _reason,
         uint256 _epoch
-    ) external onlyRole(parameters.getStakeModifierHash()) {
+    ) external onlyRole(parameters.getStakeModifierHash())
```

Comment:
Though its accessible to RewardManager only, We should pause this as well right ? WhenNotPaused() maybe wont work directly here as we want other contract functions to go on, I think we should have one if inside this block . like ``` if paused => dont change stake```

Reply:
I think this can fixed with the new influence mechanism where in the influence is changing, not the stake unless they add stake (which will be paused) or receive block reward.

Reply:
its fine. even if attacker manages to set their stake very high they will not be able to withdraw funds since withdraw/exit functions will be paused.

Reply:
hmm yes, he won't be able to withdraw 
but maybe there is a small escape here
thing is we kind of have liquid staking due to sRZR,
so I set my stake high, the value of my sRZR goes up.
and then I am free to exchange them.

Reply:
Does srzr have erc20pausabl

Reply:
Not right now 
It inherits normal ERC20 now 
We can add ERC20pausable extension.

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Pausable.sol

This will allow us complete control over a transfer .